Program.cs

```diff
@@ -24,12 +27,21 @@ private static void Main(string[] args)
         //Test methods
         app.MapGet("/", () => "Hello World!");
         app.MapGet("/test", (DataClient dataClient) => dataClient.TestConnection());
+        //Person methods
         app.MapGet("/person", (PersonService personService) => Results.Ok(personService.GetAll()));
         app.MapGet("/person/GetByName", (PersonService personService, [FromQuery] string name) => Results.Ok(personService.GetPersonByName(name)));
         app.MapGet("/person/GetByEmpType", (PersonService personService, [FromQuery] string emplType) => Results.Ok(personService.GetPersonByPersonType(emplType)));
         app.MapGet("/person/GetByNameAndType", (PersonService personService, [FromQuery] string name, string emplType) => Results.Ok(personService.GetPersonByNameAndPersonType(name, emplType)));
-        //Person methods
         //Products methods
+        app.MapGet("/product", (ProductService productService) => Results.Ok(productService.GetAll()));
+        app.MapGet("/product/GetByName", (ProductService productService, [FromQuery] string name) => Results.Ok(productService.GetProductByName(name)));
+        app.MapGet("/product/GetByCatType", (ProductService productService, [FromQuery] string categoryType) => Results.Ok(productService.GetProductByCategoryType(categoryType)));
+        //sales method
+        app.MapGet("/sales", (SalesService salesService) => Results.Ok(salesService.GetAll()));
+        app.MapGet("/sale/GetByNameAndYear", (EmployeeSalesService employeeSalesService, [FromQuery] string name, DateTime year) => 
```

Comment:
When you're working with APIs, it's important to define the hierarchical structure of the endpoints. In this case, if you start with 'sales', you should continue with 'sales', because if you want to implement Swagger in the future, Swagger can document the methods based on the entry points. For example, 'sales', 'products', 'persons', 'business', etc. You can find more information here: [link](https://blog.dreamfactory.com/best-practices-for-naming-rest-api-endpoints/)

If you're using a naming convention, be consistent. In this case, the correct way is 'GetSalesByNameAndYear'. This helps other developers easily map your methods.

When accessing this method, you need to call it like this: https://localhost:7161/sale/GetByNameAndYear?name=David&year=2011-05-31, which only returns the sales for that specific day. However, what we were asking for was the sales for the entire year.